blog/2025-08-00-uppy-5.0.mdx

```diff
@@ -0,0 +1,772 @@
+</script>
+```
+
+  </TabItem>
+  <TabItem value="svelte" label="Svelte">
+
+```html
+<UppyContextProvider {uppy}>
+	<!-- ... -->
+</UppyContextProvider>
+
+<script lang="ts">
+	import Uppy from '@uppy/core';
+	import { UppyContextProvider } from '@uppy/svelte';
+
+	const uppy = new Uppy();
+</script>
+```
+
+  </TabItem>
+</Tabs>
+
+### New components
+
+All components come with sensible defaults, but expose data attributes for
+complete styling control. This approach delivers the flexibility developers need,
+while maintaining the plug-and-play experience Uppy is known for.
+
+<Tabs>
+  <TabItem value="upload-button" label="UploadButton" default>
+
+**Attributes**
+
+- `data-uppy-element="upload-button"`
+- `data-state="init" | "uploading" | "paused" | "error" | "complete"`
+
+<UploadButton />
+
+  </TabItem>
+
+  <TabItem value="dropzone" label="Dropzone">
+
+Drag-and-drop zone component for file uploads.
+
+**Props**:
+
+- `width?: string`
+- `height?: string`
+- `note?: string`
+- `noClick?: boolean`
+
+<DropzoneDemo />
+
+  </TabItem>
+
+  <TabItem value="files-list" label="FilesList">
+
+Component for displaying a list view of selected files.
+
+**Props**:
+
+- `editFile?: (file: UppyFile<Meta, Body>) => void`
+
+<FilesListDemo />
+
+  </TabItem>
+
+  <TabItem value="files-grid" label="FilesGrid">
+
+Grid layout component for displaying files in a grid format.
+
+**Props**:
+
+- `columns?: number`
+- `editFile?: (file: UppyFile<Meta, Body>) => void`
+
+<FilesGridDemo />
+
+  </TabItem>
+
+  <TabItem value="thumbnail" label="Thumbnail">
+
+Component for displaying image thumbnails and/or SVG icons.
+
+**Props**:
+
+- `file: UppyFile<Meta, Body>`
+- `width?: string`
+- `height?: string`
+- `images?: boolean`
+
+<ThumbnailDemo />
+
+  </TabItem>
+
+  <TabItem value="provider-icon" label="ProviderIcon">
+
+Common icons, such as a webcam, device, Dropbox, Google Drive, OneDrive, etc.
+
+**Props**:
+
+- `provider`:
+  `'device' | 'camera' | 'screen-capture' | 'audio' | 'dropbox' | 'facebook' | 'instagram' | 'onedrive' | 'googlephotos' | 'googledrive'`
+- `fill?: string`
+
+<ProviderIconDemo />
+
+  </TabItem>
+
+</Tabs>
+
+#### Styling component states
+
+Components expose their internal state through data attributes, enabling precise
+CSS targeting.
+
+```css
+button[data-uppy-element='upload-button'][data-state='init'] {
+	background-color: #3b82f6;
+}
+
+button[data-uppy-element='upload-button'][data-state='uploading'] {
+	background-color: #f59e0b;
+	cursor: wait;
+}
+
+button[data-uppy-element='upload-button'][data-state='complete'] {
+	background-color: #10b981;
+}
+
+button[data-uppy-element='upload-button'][data-state='error'] {
+	background-color: #ef4444;
+}
+```
+
+### New hooks
+
+Examples are shown in React, but the same hooks are available in Vue and Svelte.
+
+<Tabs>
+  <TabItem value="use-dropzone" label="useDropzone">
+
+Here is an example that uses hooks to exactly mimic the `<Dropzone />` component.
```

Comment:
Similar to the previous comment, I think that this section with the various hook examples could be presented in a more clear and structured manner. I suggest adding small headers for each individual hook example.